lib/oli_web/live/objectives/objectives.ex

```diff
@@ -1,431 +1,447 @@
+
+  def filter_rows(socket, query, _filter) do
+    query_str = String.downcase(query)
+
+    Enum.filter(socket.assigns.objectives, fn obj ->
+      String.contains?(String.downcase(obj.title), query_str)
+    end)
+  end
 
-    objective_mappings = ObjectiveEditor.fetch_objective_mappings(project)
+  def live_path(socket, params),
+    do: Routes.live_path(socket, __MODULE__, socket.assigns.project.slug, params)
 
-    subscriptions = subscribe(objective_mappings, project.slug)
+  def mount(%{"project_id" => project_slug}, %{"current_author_id" => author_id} = _session, socket) do
+    project = Course.get_project_by_slug(project_slug)
+    author = Accounts.get_author(author_id)
 
-    objectives_tree = to_objective_tree(objective_mappings)
+    {all_objectives, all_children, objectives, table_model} =
+      build_objectives(project, [], fn socket -> socket end, true)
 
     {:ok,
-     assign(socket,
-       active: :objectives,
-       objective_mappings: objective_mappings,
-       objectives_tree: objectives_tree,
-       breadcrumbs: [Breadcrumb.new(%{full_title: "Objectives"})],
-       changeset: Resources.change_revision(%Revision{}),
-       project: project,
-       subscriptions: subscriptions,
-       attachment_summary: @default_attachment_summary,
-       modal: nil,
-       author: author,
-       force_render: 0,
-       can_delete?: true,
-       edit: :none,
-       title: "Objectives | " <> project.title
-     )}
+      assign(socket,
+        breadcrumbs: [Breadcrumb.new(%{full_title: "Objectives"})],
+        project: project,
+        author: author,
+        objectives: objectives,
+        table_model: table_model,
+        total_count: length(objectives),
+        all_objectives: all_objectives,
+        all_children: all_children
+      )
+    }
   end
 
-  def render(assigns) do
-    ~L"""
-    <%= render_modal(assigns) %>
-
-    <div class="objectives container">
-      <div class="mb-2 row">
-        <div class="col-12">
-          <p>
-            Learning objectives help you to organize course content and determine appropriate assessments and instructional strategies.
-            <br/>Refer to the <a href="https://www.cmu.edu/teaching/designteach/design/learningobjectives.html" target="_blank">CMU Eberly Center guide on learning objectives</a> to learn more about the importance of attaching learning objectives to pages and activities.
-          </p>
-        </div>
-      </div>
+  defp build_objectives(project, objectives_attachments, flash_fn, first_load \\ false) do
+    all_objectives =
+      project
+      |> ObjectiveEditor.fetch_objective_mappings()
+      |> Enum.map(& &1.revision)
+
+    all_children =
+      all_objectives
+      |> Enum.reduce([], fn rev, acc -> rev.children ++ acc end)
+      |> Enum.uniq()
+
+    objectives =
+      Enum.reduce(all_objectives, [], fn rev, acc ->
+        case Enum.find(all_children, fn child_id -> child_id == rev.resource_id end) do
```

Comment:
We should take this opportunity to remove this `find` called within `reduce`, given that this is n^2 behavior.